### PR TITLE
[debug] Add helper to run cProfile on a function

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #135257
* #135256
* #135235
* #135079
* __->__ #135084
* #135082
* #135076
* #135070

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_dynamo/debug_utils.py [hunk 1]
```diff
@@ -782,3 +784,35 @@ def gen_tensor(shape, dtype) -> Tensor:
             setattr(container, attr_name, gen_tensor(shape, dtype))
 
     return kwargs
+
+
+def profile_to_file(filename: str) -> Callable[[T], T]:
```
Comment (oulgen) [inline torch/_dynamo/debug_utils.py:789]: maybe leave a doc block explaining what it does and how to use it

although the function is short enough that maybe not necessary

**Review by oulgen — APPROVED** [review]

**Comment by laithsakka** [comment]
this stack enhanced the following benchmarks
{
  "add_loop_inductor": -1.5300736874982,
  "basic_modules_ListOfLinears_inductor": -2.1681222591775,
  "basic_modules_ModuleComparison_inductor": -2.0835290701558,
  "basic_modules_SequentialWithDuplicatedModule_inductor": -2.5733186361454
}

**Pull request closed without merging**